src/libutil/signals.hh

```diff
@@ -0,0 +1,70 @@
+#pragma once
+///@file
+
+#include "types.hh"
+#include "error.hh"
+#include "logging.hh"
+
+#include <functional>
+
+namespace nix {
+
+/* User interruption. */
+
+/**
+ * @note Does nothing on Windows
+ */
+static inline void setInterrupted(bool isInterrupted);
+
+/**
+ * @note Does nothing on Windows
+ */
+static inline bool getInterrupted();
+
+/**
+ * @note Does nothing on Windows
+ */
+static inline void setInterruptCheck(std::function<bool()> interruptCheck);
```

Comment:
GCC says
```
In file included from src/nix-collect-garbage/nix-collect-garbage.cc:2:
src/libutil/signals.hh:27:20: warning: ‘void nix::setInterruptCheck(std::function<bool()>)’ declared ‘static’ but never defined [-Wunused-function]
   27 | static inline void setInterruptCheck(std::function<bool()> interruptCheck);
      |                    ^~~~~~~~~~~~~~~~~
```

Reply:
Yeah, having static functions in a header is very weird...

Reply:
Fixed in #10411.

Reply:
Thanks, sorry about that!